everware/git_processor.py

```diff
@@ -177,3 +184,30 @@ def load_state(self, state):
             if key in state:
                 setattr(self, key, state[key])
 
+    def parse(self, everware_yml):
+        import yaml
+        with open(everware_yml) as file:
+            text = yaml.load(file)
+        volumes = []
+        for element in text:
+            volumes += text.get(element).get("volumes")
+        for volume in volumes:
+            #check url
+            self.download_file(volume)
+
+    def download_file(self, url):
+        import urllib
+        filename = url.split("/")[-1]
+        if len(filename) == 0:
+            filename = url.split("/")[-2]
+        #should think about deleting this folder someday
+        directory_volume = self._repo_dir + "-volume"
+        os.mkdir(directory_volume)
+        try:
+            urllib.request.urlretrieve(url, directory_volume + '/' + filename)
```

Comment:
I think this call is blocking, so the server will stop being responsive while doing this. Take a look at the OAuthAuthenticator and co for an example of using a non-blocking HTTP client in tornado